htmd/molecule/molecule.py

```diff
@@ -882,36 +883,23 @@ def read(self, filename, type=None, skip=None, frames=None, append=False, overwr
             readers = _ALL_READERS[ext]
             for rr in readers:
                 try:
-                    to, tr = rr(fname, frame=frame, topoloc=tmppdb)
+                    mol = rr(fname, frame=frame, topoloc=tmppdb, **kwargs)
                 except FormatError:
                     continue
                 else:
                     break
 
-            if tr is not None:
-                self._keepFrame(tr, frame)
-                self._checkCoords(tr, rr, fname)
-                # TODO: Get rid of this if by moving it to a function
-                if ext in _TRAJECTORY_READERS and frame is None:
-                    # Writing hidden index file containing number of frames in trajectory file
-                    if os.path.isfile(fname):
-                        self._writeNumFrames(fname, tr.coords[0].shape[2])
-                    ff = range(np.size(tr.coords[0], 2))
-                    #tr.step = tr.step + traj[-1].step[-1] + 1
-                elif frame is None:
-                    ff = [0]
-                elif frame is not None:
-                    ff = [frame]
-                else:
-                    raise AssertionError('Should not reach here')
-                tr.fileloc = [[fname, j] for j in ff]
-                traj += tr
+            if self.numAtoms != 0 and mol.numAtoms != self.numAtoms:
+                raise ValueError('Number of atoms in file ({}) mismatch with number of atoms in the molecule '
+                                 '({})'.format(mol.numAtoms, self.numAtoms))
 
-            if to is not None:
-                self._parseTopology(to, fname, overwrite=overwrite, _logger=_logger)
+            if frame is not None:
+                mol.dropFrames(keep=frame)
 
-        if len(traj.coords) != 0:
-            self._parseTraj(traj, skip=skip)
+            newmols.append(mol)
```

Comment:
mol can be a `list()`, so I suggest changing this to:
```python
if isinstance(mol, list):
    newmols.extend(mol)
else:
    newmols.append(mol)
```

Reply:
Good point but as we discussed Molecule should read single molecules from files. I will remove the kwargs and force the mol2 option for single molecule

Reply:
Actually I'm leaving the kwargs in case some readers in the future require it. It's a bit undocumented and hacky I guess but gives more flexibility if we want to solve some issue of a user for example and doesn't really hurt at the moment. We already have one in the .write.
mol2read by default reads only the first molecule so I think it's ok. I will add a check though that the reader didn't return multiple molecules.